autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ClusterStatusSyncHandler.java

```diff
@@ -56,6 +61,23 @@ public void onApplicationEvent(ClusterStatusSyncEvent event) {
             clusterService.setState(cluster.getId(), ClusterState.SUSPENDED);
             LOGGER.info("Suspended cluster '{}', CB Stack Status '{}', CB Cluster Status '{}'",
                     cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
+        } else if (cluster.getEnvironmentCrn() == null) {
+            String environmentCrn = cloudbreakCommunicator.getAutoscaleClusterByCrn(cluster.getStackCrn()).getEnvironmentCrn();
+            clusterService.setEnvironmentCrn(cluster.getId(), environmentCrn);
+        } else if (cluster.getMachineUserCrn() == null) {
```

Comment:
This logic should not belong to this class, the purpose of this class to execute logics based on status of the cluster, not to fill in environmentCrn for old entries in periscope. Have you checked if it is possible to fill in environmentCrn using other solutions like a migration class during application start up? Using this logic will put a high pressure on CB and autoscale at first on application startup, then application will never reach this code again.
If there is no other solution, I would suggest to refactor this code at least into another class, since this class should handle only statuses, not environmentCrns.

Reply:
for example it would be better if a separate class could call a list on datahub API, filter for relevant ones and fill the environmentCrn, in this case we would not call CB several times during application startup

Reply:
Sorry for the long gap between updates.

Addressed this by creating `InitializeMachineUserService` with `@PostConstruct` to initialize environmentCrn and machineUserCrn for older periscope clusters, so that it is executed during application startup. For new clusters, envCrn and machineUserCrn initialization is added to the `create` method which persists a newly created cluster in the persicope DB via `ClusterService`

Reply:
@aakashshankar this logic still calls CB in a foreach which can result several CB call, we should avoid that